api_tests/src/test_environment.ts

```diff
@@ -0,0 +1,119 @@
+import { Script } from "vm";
+import { Config } from "@jest/types";
+import { execSync } from "child_process";
+import commander from "commander";
+import { LedgerRunner } from "../lib/ledgers/ledger_runner";
+import { HarnessGlobal } from "../lib/utils";
+import NodeEnvironment from "jest-environment-node";
+import rimraf from "rimraf";
+import * as fs from "fs";
+import { Mutex } from "async-mutex";
+
+// ************************ //
+// Setting global variables //
+// ************************ //
+
+interface ConfigInterface {
+    ledgers: string[];
+    logDir: string;
+}
+
+export class E2ETestEnvironment extends NodeEnvironment {
+    private docblockPragmas: string;
+    private projectRoot: string;
+    private testRoot: string;
+    private logDir: string;
+    private ledgerRunner: LedgerRunner;
+    // @ts-ignore
+    public global: HarnessGlobal;
+
+    constructor(config: Config.ProjectConfig, context: any) {
+        super(config);
+
+        this.docblockPragmas = context.docblockPragmas;
+
+        this.projectRoot = execSync("git rev-parse --show-toplevel", {
```

Comment:
Please prefer the `async` variant to this function.

Reply:
Unfortunately not possible as this is in the constructor.

Reply:
You could do it all in the setup step?

Reply:
Moved it into the `setup()`.